integration/src/test/scala/com/spotify/scio/iceberg/IcebergIOIT.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2024 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.spotify.scio.iceberg
+
+import com.dimafeng.testcontainers.{ForAllTestContainer, GenericContainer}
+import com.spotify.scio.testing.PipelineSpec
+import magnolify.beam._
+import org.apache.iceberg.catalog.{Namespace, TableIdentifier}
+import org.apache.iceberg.rest.RESTCatalog
+import org.apache.iceberg.types.Types.{IntegerType, NestedField, StringType}
+import org.apache.iceberg.{CatalogProperties, CatalogUtil, PartitionSpec}
+import org.testcontainers.containers.wait.strategy.HostPortWaitStrategy
+
+import java.time.Duration
+import java.io.File
+import java.nio.file.Files
+import scala.jdk.CollectionConverters._
+
+case class IcebergIOITRecord(a: Int, b: String)
+object IcebergIOITRecord {
+  implicit val icebergIOITRecordRowType: RowType[IcebergIOITRecord] = RowType[IcebergIOITRecord]
+}
+
+class IcebergIOIT extends PipelineSpec with ForAllTestContainer {
+  val ContainerPort = 8181
+  val CatalogName = "iceberg_it"
+  val NamespaceName = "iceberg_it_ns"
+  val TableName = s"${NamespaceName}.iceberg_records"
+
+  lazy val tempDir: File = {
+    val t = Files.createTempDirectory("iceberg-it").toFile
+    t.deleteOnExit()
+    t
+  }
+
+  override val container: GenericContainer =
+    GenericContainer(
+      GenericContainer.stringToDockerImage("tabulario/iceberg-rest:1.6.0"),
+      exposedPorts = Seq(ContainerPort),
+      waitStrategy = new HostPortWaitStrategy()
+        .forPorts(ContainerPort)
+        .withStartupTimeout(Duration.ofSeconds(180))
+    )
+
+  lazy val uri = s"http://${container.containerIpAddress}:${container.mappedPort(ContainerPort)}"
+
+  override def afterStart(): Unit = {
+    val cat = new RESTCatalog()
+    cat.initialize(CatalogName, Map("uri" -> uri).asJava)
+
+    import org.apache.iceberg.Schema
+    cat.createNamespace(Namespace.of(NamespaceName))
+    cat.createTable(
+      TableIdentifier.parse(TableName),
+      new Schema(
+        NestedField.required(0, "a", IntegerType.get()),
+        NestedField.required(1, "b", StringType.get())
+      ),
```

Comment:
No, this is an Iceberg schema rather than the Beam schema. The lack of create-on-write does raise the question of whether we _also_ need to derive the iceberg schemas

Reply:
Maybe `RowType` could also offer a `def icebergSchema`? (Similar to how magnolify-parquet has both `def schema` and `def avroSchema`...)

Reply:
That would mean pulling in iceberg deps into the beam module fyi

Reply:
could make it provided, I guess, but point taken

Seems like Beam should have a utility function for converting between Beam/Icerberg Schemas. They have similar stuff for BQ/Avro/BeamSchema interop. Maybe we could contribute there

Reply:
Beam does have this but it introduces a dep on the iceberg part of the sdk that in theory should be managed.

I could use it in the integration test directly but that wouldn't help users at all.
```
IcebergUtils.beamSchemaToIcebergSchema(rowType.schema)
```